packages/apollo-graphql/src/schema/buildSchemaFromSDL.ts

```diff
@@ -237,7 +237,10 @@ export function addResolversToSchema(
       const values = type.getValues();
       const newValues: { [key: string]: GraphQLEnumValue } = {};
       values.forEach(value => {
-        const newValue = (fieldConfigs as any)[value.name] || value.name;
+        let newValue = (fieldConfigs as any)[value.name];
+        if (newValue == undefined) {
```

Comment:
This should probably be `===`

Reply:
Thanks @lilyball, I have updated. I do have a repo that shows the problem, but as you can tell it's quite a small fix.